pyhap/characteristic.py

```diff
@@ -163,70 +150,45 @@ def override_properties(self, properties=None, valid_values=None):
         if valid_values:
             self.properties['ValidValues'] = valid_values
 
-    def get_hap_value(self):
-        """Get the value of the characteristic, constrained with the HAP properties.
-        """
-        val = self.value
-        if self.properties["Format"] == HAP_FORMAT.STRING:
-            val = val[:256]
-        elif self.properties["Format"] in HAP_FORMAT.NUMERIC:
-            if "maxValue" in self.properties:
-                val = min(self.properties["maxValue"], val)
-            if "minValue" in self.properties:
-                val = max(self.properties["minValue"], val)
-        return val
-
     def notify(self):
-        """Notify clients about a value change.
-
-        .. note:: Non-blocking, i.e. does not wait for the update to be sent.
-        .. note:: Uses the `get_hap_value`, i.e. sends the HAP value.
-        .. seealso:: accessory_driver.publish
-
-        :raise NotConfiguredError: When the broker is not set.
-        """
-        if self.broker is None:
-            raise NotConfiguredError("Attempted to notify when `broker` is None. "
-                                     "Consider adding the characteristic to a "
-                                     "Service and then to an Accessory.")
-
-        data = {
-            "type_id": self.type_id,
-            "value": self.get_hap_value(),
+        """Notify clients about a value change. Sends the value."""
+        self.broker.publish(self.value, self)
+
+    def to_dict(self):
+        """Create a dict repr of this object for json serialization."""
+        json_dict = {
+            'iid': self.broker.iid_manager.get_iid(self),
+            'type': str(self.type_id).upper(),
+            'description': self.display_name,
+            'perms': self.properties['Permissions'],
+            'format': self.properties['Format'],
         }
-        self.broker.publish(data, self)
 
-    def to_HAP(self, iid_manager):
-        """Create a HAP representation of this Characteristic.
+        if self.properties['Format'] in HAP_FORMAT.NUMERIC:
+            json_dict.update({k: self.properties[k] for k in
+                              self.properties.keys() & _HAP_NUMERIC_FIELDS})
+        elif self.properties['Format'] == HAP_FORMAT.STRING:
+            if len(self.value) > 64:
+                json_dict['maxLen'] = min(len(self.value), 256)
+        if HAP_PERMISSIONS.READ in self.properties['Permissions']:
+            json_dict['value'] = self.value
 
-        .. note:: Uses the `get_hap_value`, i.e. sends the HAP value.
+        return json_dict
 
-        :param iid_manager: IID manager to query for this object's IID.
-        :type iid_manager: IIDManager
+    def update_value(self, value):
```

Comment:
How about `client_set_value` to make it more clear?

Reply:
Sounds good :+1:

Reply:
Could I suggest `client_change_value`, I think it makes it clearer that I might fail. But I could just be bikeshedding.

Reply:
What about `client_update_value`?